src/components/appheader.main.jsx

```diff
@@ -127,9 +128,11 @@ class AppHeaderMain extends React.Component {
             </div>
           </div>
 
-          <div className="search-container">
-            <AppHeaderSearchMain isMobileView={false} />
-          </div>
+          {!isB2B && !Config.b2b.enable && (
```

Comment:
This condition won't work as expected. We need to show these components only when we're in B2C, or only in B2B after a buyer has logged in.

Something like the following should work better:
`{!Config.b2b.enable || (Config.b2b.enable && isB2B) && (....`